include/boost/url/decoded_view.hpp

```diff
@@ -0,0 +1,1049 @@
+
+        @return Number of characters copied
+    */
+    BOOST_CXX14_CONSTEXPR
+    size_type
+    copy(char* dest,
+         size_type count,
+         size_type pos = 0) const;
+
+    /** Compares two strings
+
+        The length of the sequences to compare is the smaller of
+        `size()` and `other.size()`.
+
+        The function compares the two strings as if by calling
+        `char_traits<char>::compare(to_string().data(), v.data(), rlen)`.
+
+        @param other string to compare
+
+        @return Negative value if this string is less than the other
+        character sequence, zero if the both character sequences are
+        equal, positive value if this string is greater than the other
+        character sequence
+    */
+    BOOST_CXX14_CONSTEXPR
+    int
+    compare(string_view other) const noexcept;
+
+    /// @copydoc compare()
+    BOOST_CXX14_CONSTEXPR
+    int
+    compare(char const* other) const
+    {
+        return compare(string_view(other));
+    }
+
+    /** Compares two strings
+
+        The length of the sequences to compare is the smaller of
+        `size()` and `other.size()`.
+
+        The function compares the two strings as if by calling
+        `char_traits<char>::compare(to_string().data(), v.to_string().data(), rlen)`.
+
+        @param other string to compare
+
+        @return Negative value if this string is less than the other
+        character sequence, zero if the both character sequences are
+        equal, positive value if this string is greater than the other
+        character sequence
+    */
+    BOOST_CXX14_CONSTEXPR
+    int
+    compare(decoded_view other) const;
+
+    /** Returns a basic_string
+
+        This overload allows the caller to recycle
+        the input string.
+
+        @return A string representing the
+        entire contents of the decoded range.
+    */
+    template <class A>
+    BOOST_URL_CXX20_CONSTEXPR
+    std::basic_string<char, std::char_traits<char>, A>&
+    to_string(std::basic_string<char, std::char_traits<char>, A> & str) const
+    {
+        str.resize(size());
+        copy(&str[0], size());
+        return str;
+    }
+
+    /** Returns a basic_string
+
+        This overload constructs a new string.
+
+        @return A string representing the
+        entire contents of the decoded range.
+    */
+    template <class A = std::allocator<char>>
+    BOOST_URL_CXX20_CONSTEXPR
+    std::basic_string<char, std::char_traits<char>, A>
+    to_string() const
+    {
+        std::basic_string<
+            char, std::char_traits<char>, A> str(size(), ' ');
+        return to_string(str);
+    }
+
+    /** Returns a basic_string
+
+        @return A string representing the
+        entire contents of the decoded range.
+    */
+    template <class A>
+    BOOST_URL_CXX20_CONSTEXPR explicit
+    operator std::basic_string<char, std::char_traits<char>, A>() const
+    {
+        return to_string<A>();
```

Comment:
This function should be removed, since `assign_to` does the same job, But I will also point out that this function is broken, it only works when `A` is default constructible, which is almost never the case for custom allocators.